stores/navigationStore.ts

```diff
@@ -0,0 +1,35 @@
+import { create } from 'zustand';
+import type { Pages } from '../types';
+
+interface NavigationState {
+  prev_navigation_states: Pages[];
+  current_navigation_state: Pages;
+  updateNavigation: (page: Pages) => void;
+  revertToPreviousState: () => void;
+}
+
+export const useNavigationStore = create<NavigationState>((set) => ({
+  prev_navigation_states: [] as Pages[],
+  current_navigation_state: 'Login',
+  updateNavigation: (page) => {
+    set((state) => ({
+      ...state,
+      current_navigation_state: page,
+      prev_navigation_states:
+        state.prev_navigation_states.length >= 5
+          ? [page, ...state.prev_navigation_states.slice(0, 4)]
```

Comment:
This doesn't seem necessary

Reply:
I want to restrict to just last 5 pages visited